libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookHelper.cs

```diff
@@ -167,6 +174,19 @@ public static List<Attachment> HandleMessageAttachments(Message message)
             return attachmentsList;
         }
 
+        /// <summary>
+        /// Sends a message that will display a typing indicator as a placeholder while the message arrives.
+        /// </summary>
+        /// <param name="activity">The activity with the actual message to be sent.</param>
+        /// <param name="turnContext">The TurnContext that will send the activities.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>A task.</returns>
+        public static async Task SendActivityWithTypingIndicatorAsync(IActivity activity, ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            var typingActivity = GenerateTypingActivity(turnContext.Activity.Conversation.Id);
+            await turnContext.SendActivitiesAsync(new[] { typingActivity, activity }, cancellationToken).ConfigureAwait(false);
```

Comment:
I think this might be extraneous since the messages will both go out very quickly in sequence meaning there won't be much time for the typing indicator to be displayed before the next message arrives...

Reply:
Hi Ben. Then we'll change it to let the user manually create the _typing indicator_ activity and then send it before the _message_ activity (basically what the method **SendActivityWithTypingIndicatorAsync** is doing. We would then remove the method of course).
We will also include an example with a dialog.

Reply:
that sounds right to me!

Reply:
Ok now **SendActivityWithTypingIndicatorAsync** has been removed. The bot needs to create a typing indicator activity by calling **FacebookHelper.GenerateTypingActivity**. 
There are examples for sending messages with and without typing indicator in the bot, as well as a trigger phrase (`display typing indicator`) for displaying a typing indicator alone.
I also made some other minor corrections I found.